src/pdef.F

```diff
@@ -341,36 +334,39 @@ subroutine pdefy1(vh,rho0,gz,rgz,rrv,advy,dum,mass,s0,s,dt,flag,south,newsouth,n
         enddo
         enddo
         enddo
+        !$acc end parallel
       ENDIF
 
         if(sbc.eq.2 .and. ibs.eq.1)then
           j=1
           !$omp parallel do default(shared) private(i,k)
-          !$acc parallel loop gang collapse(2) private(i,k)
+          !$acc parallel loop gang vector collapse(2) default(present)
           do k=1,nk
           do i=1,ni
             if(rrv(i,j,k).ge.0.0)then
               dum(i,j,k)=rho0(i,j,k)*s0(i,j,k)
             endif
           enddo
           enddo
+          !$acc end parallel
         endif
 
         if(nbc.eq.2 .and. ibn.eq.1)then
           j=nj
           !$omp parallel do default(shared) private(i,k)
-          !$acc parallel loop gang collapse(2) private(i,k)
+          !$acc parallel loop gang vector collapse(2) default(present)
           do k=1,nk
           do i=1,ni
             if(rrv(i,nj+1,k).le.0.0)then
               dum(i,j,k)=rho0(i,j,k)*s0(i,j,k)
             endif
           enddo
           enddo
+          !$acc end parallel
```

Comment:
@sjsprecious you are inconsistent in your addition of the '$acc end parallel' clause. Sometimes it is present while others it is not.

Reply:
ok